NDTensors/src/lib/Sectors/src/abstractcategory.jl

```diff
@@ -1,45 +1,129 @@
+  return GradedAxes.gradedrange([trivial(eltype(axis_type))])  # always returns nondual
+end
+function trivial(la_type::Type{<:LabelledNumbers.LabelledInteger})
+  return la_type(1, trivial(LabelledNumbers.label_type(la_type)))
+end
+function trivial(type::Type)
+  return error("`trivial` not defined for type $(type).")
 end
 
-function label_fusion_rule(category_type::Type{<:AbstractCategory}, l1, l2)
-  return error("`label_fusion_rule` not defined for type $(category_type).")
+istrivial(c::AbstractCategory) = (c == trivial(c))
+
+function category_label(c::AbstractCategory)
+  return error("method `category_label` not defined for type $(typeof(c))")
 end
 
-function fusion_rule(c1::AbstractCategory, c2::AbstractCategory)
-  category_type = typeof(c1)
-  return [category_type(l) for l in label_fusion_rule(category_type, label(c1), label(c2))]
+block_dimensions(g::AbstractUnitRange) = block_dimensions(SymmetryStyle(g), g)
+block_dimensions(::AbelianGroup, g) = GradedAxes.unlabel.(BlockArrays.blocklengths(g))
+function block_dimensions(::SymmetryStyle, g)
+  return Sectors.quantum_dimension.(GradedAxes.blocklabels(g)) .*
+         BlockArrays.blocklengths(g)
 end
 
+quantum_dimension(x) = quantum_dimension(SymmetryStyle(x), x)
+
+function quantum_dimension(::SymmetryStyle, c::AbstractCategory)
+  return error("method `quantum_dimension` not defined for type $(typeof(c))")
+end
+
+quantum_dimension(::AbelianGroup, ::AbstractCategory) = 1
+quantum_dimension(::EmptyCategory, ::AbstractCategory) = 1
+quantum_dimension(::SymmetryStyle, g::AbstractUnitRange) = sum(block_dimensions(g))
+quantum_dimension(::AbelianGroup, g::AbstractUnitRange) = length(g)
+
+# ===============================  Fusion rule interface  ==================================
 ⊗(c1::AbstractCategory, c2::AbstractCategory) = fusion_rule(c1, c2)
 
-⊕(c1::AbstractCategory, c2::AbstractCategory) = [c1, c2]
-⊕(cs::Vector{<:AbstractCategory}, c::AbstractCategory) = [cs; c]
-⊕(c::AbstractCategory, cs::Vector{<:AbstractCategory}) = [c; cs]
+function fusion_rule(c1, c2)
+  return fusion_rule(combine_styles(SymmetryStyle(c1), SymmetryStyle(c2)), c1, c2)
+end
 
-function Base.show(io::IO, cs::Vector{<:AbstractCategory})
-  (length(cs) <= 1) && print(io, "[")
-  symbol = ""
-  for c in cs
-    print(io, symbol, c)
-    symbol = " ⊕ "
-  end
-  (length(cs) <= 1) && print(io, "]")
-  return nothing
+function fusion_rule(::SymmetryStyle, c1::C, c2::C) where {C<:AbstractCategory}
+  degen, labels = label_fusion_rule(C, category_label(c1), category_label(c2))
+  return GradedAxes.gradedrange(LabelledNumbers.labelled.(degen, C.(labels)))
 end
 
-function trivial(category_type::Type{<:AbstractCategory})
-  return error("`trivial` not defined for type $(category_type).")
+# abelian case: return Category
+function fusion_rule(::AbelianGroup, c1::C, c2::C) where {C<:AbstractCategory}
+  return C(label_fusion_rule(C, category_label(c1), category_label(c2)))
```

Comment:
See my comment below about how I don't like that `label_fusion_rule` outputs different formats depending on the symmetry style.